src/shogun/classifier/LDA.cpp

```diff
@@ -175,38 +192,38 @@ bool CLDA::train_machine(CFeatures *data)
 		//+ve covariance matrix
 #if EIGEN_WITH_OPERATOR_BUG
 		cen_pos=cen_pos*cen_pos.transpose()
-		cen_pos/=float64_t(num_pos-1);
+		cen_pos/=(ST)(num_pos-1);
 #else
-		cen_pos=cen_pos*cen_pos.transpose()/(float64_t(num_pos-1));
+		cen_pos=cen_pos*cen_pos.transpose()/((ST)(num_pos-1));
 #endif
 
 		//-ve covariance matrix
 #if EIGEN_WITH_OPERATOR_BUG
 		cen_neg=cen_neg=cen_neg*cen_neg.transpose()
-		cen_neg/=float64_t(num_neg-1);
+		cen_neg/=(ST)(num_neg-1);
 #else
-		cen_neg=cen_neg*cen_neg.transpose()/(float64_t(num_neg-1));
+		cen_neg=cen_neg*cen_neg.transpose()/((ST)(num_neg-1));
 #endif
 
 		//within class matrix
-		MatrixXd Sw= num_pos*cen_pos+num_neg*cen_neg;
-		float64_t trace=Sw.trace();
-		double s=1.0-m_gamma;
+		typename SGMatrix<ST>::EigenMatrixXt Sw= num_pos*cen_pos+num_neg*cen_neg;
+		ST trace=Sw.trace();
+		ST s=1.0-((ST)m_gamma);
 		Sw *=s;
-		Sw.diagonal()+=VectorXd::Constant(num_feat, trace*m_gamma/num_feat);
+		Sw.diagonal()+=Eigen::DenseBase<typename SGVector<ST>::EigenVectorXt>::Constant(num_feat, trace*((ST)m_gamma)/num_feat);
```

Comment:
@vigsterkr As far as I know, there is no VectorXt in Eigen - it'd be nice to have that type.  I can create that in another PR.